### PR TITLE
Refactor CLI code for easier reuse

### PR DESCRIPTION
**Pull request by bajtos**
While working on https://github.com/strongloop/loopback-next/pull/3688 (`lb4 import-lb3-model`), I discovered many obstacles preventing me from reusing existing code in my new generator.

This pull request contains a series of commits to make various bits in our CLI code base easier to use in new ways.

It may be best to review the changes commit-by-commit.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/cli/lib/utils.js [hunk 1]
```diff
@@ -103,6 +104,32 @@ exports.validateClassName = function(name) {
   return util.format('Class name is invalid: %s', name);
 };
 
+exports.remindAboutNamingRules = function(name, log) {
+  if (name.includes('_')) {
+    this.log(
```
Comment (raymondfeng) [inline packages/cli/lib/utils.js]: this.log => log?

File: packages/cli/lib/utils.js [hunk 2]
```diff
@@ -103,6 +104,32 @@ exports.validateClassName = function(name) {
...
+      chalk.red('>>> ') +
+        `Underscores _ in the class name will get removed: ${name}`,
+    );
+  }
+  if (name.match(/[\u00C0-\u024F\u1E00-\u1EFF]/)) {
+    this.log(
```
Comment (raymondfeng) [inline packages/cli/lib/utils.js]: this.log => log?

File: packages/cli/lib/utils.js [hunk 3]
```diff
@@ -103,6 +104,32 @@ exports.validateClassName = function(name) {
   return util.format('Class name is invalid: %s', name);
 };
 
+exports.remindAboutNamingRules = function(name, log) {
```
Comment (raymondfeng) [inline packages/cli/lib/utils.js]: Is `checkNamingConvention` a better name?

File: packages/cli/lib/utils.js [hunk 4]
```diff
@@ -103,6 +104,32 @@ exports.validateClassName = function(name) {
...
+  }
+  if (name.match(/[\u00C0-\u024F\u1E00-\u1EFF]/)) {
+    this.log(
+      chalk.red('>>> ') +
+        `Accented chars in the class name will get replaced: ${name}`,
+    );
+  }
+};
+
+exports.printClassFileName = function(type, typePlural, name, log) {
```
Comment (raymondfeng) [inline packages/cli/lib/utils.js]: printClassFileName => logClassGeneration?

File: packages/cli/lib/base-generator.js [hunk 5]
```diff
@@ -444,4 +445,34 @@ module.exports = class BaseGenerator extends Generator {
     }
     await this._runLintFix();
   }
+
+  // Check all files being generated to ensure they succeeded
+  _wasGenerationSuccessful() {
```
Comment (raymondfeng) [inline packages/cli/lib/base-generator.js]: Let's use present tense to be consistent: `_isGenerationSuccessful`

File: packages/cli/lib/utils.js (hunk 3 above)
Comment (bajtos) [reply packages/cli/lib/utils.js]: Not in my mind. I understand `checkNamingConvention` to report an error when the name does not match certain pattern(s). This function just prints a warning to explain the user what will happen.

File: packages/cli/lib/utils.js (hunk 4 above)
Comment (bajtos) [reply packages/cli/lib/utils.js]: I'll go with `logClassCreation` to match the message printed. I find `generation` ambiguous - is it describing the process of generating something new, or a generation like a group of similarly-aged organisms?

File: packages/cli/lib/utils.js (hunk 3 above)
Comment (raymondfeng) [reply packages/cli/lib/utils.js]: Maybe `reportNamingIssues` or `warnInvalidName`?

Comment (agnes512) [reply packages/cli/lib/utils.js]: `warnInputNaming`?

Comment (bajtos) [reply packages/cli/lib/utils.js]: I'll go with `logNamingIssues` for consistency with `logClassCreation`.

File: packages/cli/generators/model/property-definition.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
+// Copyright IBM Corp. 2018,2019. All Rights Reserved.
```
Comment (nabdelgadir) [inline packages/cli/generators/model/property-definition.js]: should just be `2019`

File: packages/cli/test/integration/lib/check-loopback-project.suite.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
```
Comment (nabdelgadir) [inline packages/cli/test/integration/lib/check-loopback-project.suite.js]: `2019` :)

**Pull request merged**